AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java

```diff
@@ -2722,24 +2725,66 @@ public void onExtendStudyLimits() {
     }
 
     public void handleEmptyCards() {
-        CollectionTask.launchCollectionTask(FIND_EMPTY_CARDS, handlerEmptyCardListener());
+        mEmptyCardTask = CollectionTask.launchCollectionTask(FIND_EMPTY_CARDS, handlerEmptyCardListener());
     }
     private HandleEmptyCardListener handlerEmptyCardListener() {
         return new HandleEmptyCardListener(this);
     }
     private static class HandleEmptyCardListener extends TaskListenerWithContext<DeckPicker> {
+        private final int mNumberOfCards;
+        private long cardSeen = 0;
+
         public HandleEmptyCardListener(DeckPicker deckPicker) {
             super(deckPicker);
+            mNumberOfCards = deckPicker.getCol().cardCount();
+        }
+
+        private void confirmCancel(DeckPicker deckPicker, CollectionTask task) {
+            new MaterialDialog.Builder(deckPicker)
+                    .content(deckPicker.getResources().getString(R.string.confirm_cancel))
+                    .positiveText(deckPicker.getResources().getString(R.string.yes))
+                    .negativeText(deckPicker.getResources().getString(R.string.no))
+                    .onNegative((x, y) -> actualOnPreExecute(deckPicker))
+                    .onPositive((x, y) -> task.safeCancel()).show()
+                    ;
         }
 
         @Override
         public void actualOnPreExecute(@NonNull DeckPicker deckPicker) {
+            DialogInterface.OnCancelListener onCancel = (dialogInterface) -> {
+                CollectionTask mEmptyCardTask = deckPicker.mEmptyCardTask;
+                if (mEmptyCardTask != null) {
+                    confirmCancel(deckPicker, mEmptyCardTask);
+                }};
             deckPicker.mProgressDialog = StyledProgressDialog.show(deckPicker, "",
-                    deckPicker.getResources().getString(R.string.emtpy_cards_finding), false);
+                    deckPicker.getResources().getString(R.string.emtpy_cards_finding), true,
+                    onCancel);
+            deckPicker.mProgressDialog.setCanceledOnTouchOutside(false);
         }
 
         @Override
-        public void actualOnPostExecute(@NonNull DeckPicker deckPicker, TaskData result) {
+        public void actualOnProgressUpdate(@NonNull DeckPicker deckPicker, @NonNull TaskData progress) {
+            cardSeen += progress.getInt();
+            int percent = Math.min(((int) (cardSeen * 100) / mNumberOfCards), 100);
+            deckPicker.mProgressDialog.setContent(
+                                                  deckPicker.getResources().getString(R.string.empty_cards_finding_percent, percent) + cardSeen + " / " + mNumberOfCards);
```

Comment:
I am also usually trying to do the minimum UI too, I will admit, this one in particular I was thinking percent *only* with no other info, like so: 
```suggestion
                                                  deckPicker.getResources().getString(R.string.empty_cards_finding_percent, percent));
```

That's how we avoid all the RTL language business - we can't do string concatenation in general because you never know where the language puts numbers naturally when doing translations. This way we have a percent and that's it

Reply:
Done. That was only an accident the remaining remained. It was already removed while you wrote this